documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/sessions/session-sessioncache.adoc

```diff
@@ -14,30 +14,30 @@
 [[pg-server-session-cache]]
 ==== The SessionCache
 
-There is one `SessionCache` per `SessionHandler`, and thus one per context.
-Its purpose is to provide an L1 cache of `Session` objects.
-Having a working set of `Session` objects in memory allows multiple simultaneous requests for the same session to share the same `Session` object.
+There is one `SessionCache` per `SessionManager`, and thus one per context.
+Its purpose is to provide an L1 cache of `ManagedSession` objects.
+Having a working set of `ManagedSession` objects in memory allows multiple simultaneous requests for the same session (ie the _same_ session id in the _same_ context) to share the same `ManagedSession` object.
 A `SessionCache` uses a `SessionDataStore` to create, read, store and delete the `SessionData` associated with the `Session`.
 
-There are two ways to create a `SessionCache` for a `SessionHandler`:
+There are two ways to create a `SessionCache` for a `SessionManager`:
 
-. allow the `SessionHandler` to create one lazily at startup.
-The `SessionHandler` looks for a `SessionCacheFactory` bean on the server to produce the `SessionCache` instance.
-It then looks for a `SessionDataStoreFactory` bean on the server to produce a `SessionDataStore` instance to use with the `SessionCache`.
+. allow the `SessionManager` to create one lazily at startup.
+The `SessionManager` looks for a `SessionCacheFactory` bean on the `Server` to produce the `SessionCache` instance.
+It then looks for a `SessionDataStoreFactory` bean on the `Server` to produce a `SessionDataStore` instance to use with the `SessionCache`.
 
-. pass a fully configured `SessionCache` instance to the `SessionHandler`.
+. pass a fully configured `SessionCache` instance to the `SessionManager`.
 You are responsible for configuring both the `SessionCache` instance and its `SessionDataStore` 
 
 More on ``SessionDataStore``s xref:pg-server-session-datastore[later], in this section we will concentrate on the `SessionCache` and `SessionCacheFactory`.
 
 
-The link:{javadoc-url}/org/eclipse/jetty/server/session/AbstractSessionCache.html[AbstractSessionCache] provides most of the behaviour of ``SessionCache``s.
+The link:{javadoc-url}/org/eclipse/jetty/session/AbstractSessionCache.html[AbstractSessionCache] provides most of the behaviour of ``SessionCache``s.
 If you are implementing a custom `SessionCache` we strongly recommend you extend this base class, as the Servlet Specification has many subtleties and extending the base class ensures that your implementation will take account of them.
```

Comment:
```suggestion
If you are implementing a custom `SessionCache` you should extend this base class, as the Servlet specification has many subtleties, and extending the base class ensures that your implementation will account for them.
```

Reply:
I've modified the wording a little differently.